Library/Homebrew/cmd/update.sh

```diff
@@ -378,9 +385,14 @@ EOS
   if [[ -n "$HOMEBREW_UPDATED" || -n "$HOMEBREW_UPDATE_FAILED" ]]
   then
     brew update-report "$@"
-    return $?
+    exitcode=$?
+    unlock update
+    return $exitcode
   elif [[ -z "$HOMEBREW_UPDATE_PREINSTALL" ]]
   then
     echo "Already up-to-date."
+    unlock update
+  else
+    unlock update
```

Comment:
However, we have different trap used in update. And bash doesn't support nested trap, right?